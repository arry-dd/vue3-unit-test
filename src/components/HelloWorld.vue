<template>
  <h1 class="name">{{ name }}</h1>
</template>

<script setup>
import { ref } from 'vue';
const name = ref('hello world!');
</script>

<style scoped>
</style>
