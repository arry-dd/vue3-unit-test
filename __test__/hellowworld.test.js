import {
  describe, expect, it
} from 'vitest';
import { mount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';

describe('测试示例', () => {
  it('hellow world', () => {
    const wrapper = mount(HelloWorld);
    expect(wrapper.html()).toMatchSnapshot(); // 快照测试
    expect(wrapper.find('.name').text()).toBe('hello world!');
  });
});
