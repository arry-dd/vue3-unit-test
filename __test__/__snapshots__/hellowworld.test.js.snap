// Vitest Snapshot v1

exports[`测试示例 > hellow world 1`] = `"<h1 class=\\"name\\">hello world!</h1>"`;
